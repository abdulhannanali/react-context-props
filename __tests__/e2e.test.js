import React from 'react'
import PropTypes from 'prop-types'
import { mount } from 'enzyme'

import { getContextualizer, withPropsFromContext } from '../src/index.js'

const Text = ({ label, color }) => <p style={{ color }}>{label}</p>

const Contextualizer = getContextualizer({ color: PropTypes.string })
const TextThatGetsColorFromContext = withPropsFromContext(['color'])(Text)

const getColorContextualizer = color => (
  <Contextualizer color={color}>
    <TextThatGetsColorFromContext label='Lorem ipsum' />
    <TextThatGetsColorFromContext label='Dolor sit amet' />
  </Contextualizer>
)

describe('End to End Context Color', () => {
  test('`color` prop in Contextualizer is green', () => {
    const color = 'green'
    const GreenContext = getColorContextualizer(color)

    const wrapper = mount(GreenContext)

    expect(wrapper.html()).toMatchSnapshot()
    expect(wrapper.find(Text).at(0).prop('color')).toBe(color)
    expect(wrapper.find(Text).at(1).prop('color')).toBe(color)
  })

  test('`color` prop in Contextualizer is red', () => {
    const color = 'red'
    const RedContext = getColorContextualizer(color)

    const wrapper = mount(RedContext)

    expect(wrapper.html()).toMatchSnapshot()
    expect(wrapper.find(Text).at(0).prop('color')).toBe(color)
    expect(wrapper.find(Text).at(1).prop('color')).toBe(color)
  })
})
