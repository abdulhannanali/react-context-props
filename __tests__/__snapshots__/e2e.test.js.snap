// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`End to End Context Color \`color\` prop in Contextualizer is green 1`] = `"<span><p style=\\"color: green;\\">Lorem ipsum</p><p style=\\"color: green;\\">Dolor sit amet</p></span>"`;

exports[`End to End Context Color \`color\` prop in Contextualizer is red 1`] = `"<span><p style=\\"color: red;\\">Lorem ipsum</p><p style=\\"color: red;\\">Dolor sit amet</p></span>"`;
